process/sync/baseSync.go

```diff
@@ -142,6 +141,10 @@ func (boot *baseBootstrap) requestedHeaderHash() []byte {
 }
 
 func (boot *baseBootstrap) processReceivedHeader(headerHandler data.HeaderHandler, headerHash []byte) {
+	if boot.shardCoordinator.SelfId() != headerHandler.GetShardID() {
```

Comment:
why did we changed the logic in baseBootstrap?

Reply:
because on sync we must process header that are for our shard